xls/public/BUILD

```diff
@@ -255,3 +255,75 @@ cc_test(
         "@com_google_googletest//:gtest",
     ],
 )
+
+cc_binary(
+    name = "libxls.dylib",
+    linkshared = True,
+    deps = [
+        ":c_api",
+    ],
+    copts = [
+        "-fno-exceptions",
+    ],
+    linkopts = [
+        "@$(location :generate_linker_params_underscores)",
+    ],
+    additional_linker_inputs = [
+        ":exported_symbols_map",
+        ":generate_linker_params_underscores",
+    ],
+    target_compatible_with = select({
+        "@platforms//os:osx": [],
+        "//conditions:default": ["@platforms//:incompatible"],
+    }),
+)
+
+# This allows export of all symbols that start with `xls_`
+filegroup(
+    name = "exported_symbols_map",
+    srcs = ["exported_symbols.map"],
+)
+
+py_test(
+    name = "test_c_api_symbols",
+    srcs = ["test_c_api_symbols.py"],
+    data = [
+        "c_api_symbols.txt",
+        ":c_api",
+        ":c_api_dslx",
+        ":c_api_vast",
+    ],
+)
+
+# We have to note that all these symbols, though unused, should be
+# included/exposed in the resulting artifact's symbol set.
+genrule(
+    name = "generate_linker_params",
+    srcs = ["c_api_symbols.txt"],
+    outs = ["dso_linker_params.txt"],
+    cmd = "sed 's|^|-Wl,-u,|' $(location c_api_symbols.txt) > $@",
+)
+
+# As above, but with leading underscores for OS X.
+genrule(
+    name = "generate_linker_params_underscores",
+    srcs = ["c_api_symbols.txt"],
+    outs = ["dso_linker_params_underscores.txt"],
+    cmd = "sed 's|^|-Wl,-u,_|' $(location c_api_symbols.txt) > $@",
```

Comment:
Interesting, I had seen that `sed` on OSx does not allow the gnu extensions of using pipe-symbol instead of slash, (as they ship ancient Unix tools) but apparently this is working well in the bazel-supplied tool ?
(no request to change, just a remark)

Reply:
This is one of these situations where somebody who is a wine connoisseur is like "I'm experiencing faint notes of chestnut" and I'm like, "Yeah, tastes like red". :laughing:

Just greedily tried to find ways to make sed do what I wanted I don't know it deeply, I pretty quickly switch to a Perl one liner. :grimacing: